script.lua

```diff
@@ -7729,122 +7732,82 @@ function onObjectDestroy(obj)
-                    },
-                    Nickname = player_color.."'s "..card.getName().." Reminder",
-                    Tags = {"Destroy", "Mask", "Reminder Token"},
-                    Grid = false,
-                    Snap = false,
-                    CustomMesh = {
-                        MeshURL = "http://cloud-3.steamusercontent.com/ugc/1749061746121830431/DE000E849E99F439C3775E5C92E327CE09E4DB65/",
-                        DiffuseURL = "http://cloud-3.steamusercontent.com/ugc/2050879298352687582/0903B5F8D08AB12D8F4C05A703A9E193F049A702/",
-                        MaterialIndex = 1
-                    },
-                    LuaScript = "function onLoad()Wait.frames(function() self.UI.setXml(self.UI.getXml()) end, 2)end",
-                    XmlUI = "<Panel rotation=\"0 0 180\" width=\"350\" height=\"490\" position=\"31 86 -11\"><Mask image=\"CardMask\"><Image image=\""..imageURL.."\"/></Mask></Panel>",
-                }
-                if Tints[player_color] then
-                    local color
-                    if Tints[player_color].Token then
-                        color = Color.fromHex(Tints[player_color].Token)
-                    else
-                        color = Color.fromHex(Tints[player_color].Presence)
-                    end
-                    data.ColorDiffuse = {
-                        r = color.r,
-                        g = color.g,
-                        b = color.b
-                    }
-                end
-                spawnObjectData({data = data})
-            end,
-            false)
-    end
+    return nil
 end
-function spawnSpiritMarker(player_color, spirit)
-    local color = getSpiritColor({name = spirit.getName()})
-    if not color then
-        color = player_color
+function spawnMaskedReminder(color, obj, isMarker)
+    local objData = obj.getData()
+    local position = obj.getPosition()
+    local imageURL, maskImage
+    local panelWidth, panelHeight, panelX, panelY
+    local name, tags, scale
+
+    if obj.hasTag("Spirit") and objData.Name == "Custom_Tile" then
+        local spiritColor = getSpiritColor({name = obj.getName()})
+        if spiritColor then
+            color = spiritColor
+        end
+
+        position = position + Vector(0, 0.5, 0)
+        imageURL = objData.CustomImage.ImageSecondaryURL
+        if obj.hasTag("Lower Spirit Image") then
+            panelWidth, panelHeight, panelX, panelY = 378, 252, -97, -33
+            maskImage = "SpiritMask2"
+        else
+            panelWidth, panelHeight, panelX, panelY = 480, 320, -127, 46
+            maskImage = "SpiritMask"
+        end
+    elseif obj.type == "Card" then
+        position = position + Vector(0, 0.2, 0)
+        imageURL = getCardImageURL(obj)
+        panelWidth, panelHeight, panelX, panelY = 350, 490, 31, 86
+        maskImage = "CardMask"
+    else
+        return
+    end
+
+    if imageURL == nil then
+        return
+    end
+
+    if isMarker then
```

Comment:
could we just drop the `isMarker` parameter and move this into the above conditional blocks?

Reply:
Whether or not we want a marker isn't a property of the object we're spawning it for: if we're acting on a spirit panel, we might be asking for a reminder or for a spirit marker. So this definitely has to be a parameter.

I considered having the function abort if it was called on a card with `isMarker = true`, but the code would still work without errors in this case, so there's no real reason to forbid it (it just never gets called that way currently).

Reply:
ah sorry the diff for this PR is not very pretty, need to have like 4 windows up to see what is really going on here. i'll have to take another look at this when it's not super late xD

Reply:
Yeah, sorry about the diff; I noticed that myself when I was adding it. Really what I did was write one new function and turn a large chunk inside three other functions into just calls to that function. I'll see if I can find a way to make it display the diff like that for easier reading.

Reply:
If you pull this branch locally and run `git diff --anchored="function applyPowerCardContextMenuItems(card)" beta reminder-hotkey`, then it'll display the diff in a *useful* format. I don't know if there's any way to make GitHub do the same.